pkgsettings/pkgsettings.py

```diff
@@ -41,8 +76,15 @@ def configure(self, obj=None, **kwargs):
             for key, new_value in kwargs.items():
                 setattr(obj, key, new_value)
 
+        if obj is self:
```

Comment:
I'm not so sure.

1. The primary goal is to warn the user when setting up the configuration so it's found as soon as possible
1. Having `self` in there once will already lead to recursion, so `self` should never be in the chain.
  1. We can actually stop from adding self to the chain, so that's what I did (and issue a warning).
1. Having object C inside the chain of object B which is inside the chain of object A is ok, but if you add A inside of C, this will result in infinite recursion again. We could actually stop this, but then the code would be much more complex.
  1. So it's not only about having `self` in the chain, it's really about loops.

So to detect these two, you should be checking for:

1. is one item added more than once in the whole graph,
1. is `self` also present somewhere (a special case, since technically it's part of the chain, but not explicitly)

Note:
Having the same settings object in the chain twice is not really a blocker, since it doesn't have to be a loop, it can also be just inefficiency. Having self in the loop definitely creates a loop.

That's why I have these two cases.